### PR TITLE
feature/issue 24

### LINKED ISSUE #24: [Feature] Search/ Update UI
### Overview
- // Update UI

### Impact
- // Search screen

### Screenshots (Optional)
- // Capture affected screens

### PR DESCRIPTION
**Pull request by dong2810**
## Prerequisite
- [x] Self-review :monocle_face:
- [x] No Crashes ☠️
---

### Root cause
- // Update UI

### Solution
- // Update UI

### Impact
- // Screen

### Screenshots (Optional)
- // Capture aff
Simulator Screen Shot - iPhone 12 Pro - 2021-11-09 at 16 29 21
ected screens
- 
Simulator Screen Shot - iPhone 12 Pro - 2021-11-09 at 16 52 05

### CODE REVIEW

**Comment by dong2810** [comment]
complete screen search

File: MovieApp/Networking/MovieAppResourceResponseAdapter.swift [hunk 1]
```diff
@@ -32,6 +32,21 @@ extension MovieAppResourceResponseAdapter: INetworkErrorResponseHandler {
...
+			case .success(let resources):
+				let movies = resources.compactMap({
+					MovieModel(movies: $0)
+				})
+				completion(.success(movies))
+			case .failure(let error):
+				completion(.failure(error))
+			}
+		}
+
```
Comment (namnh92) [inline MovieApp/Networking/MovieAppResourceResponseAdapter.swift]:
```suggestion
```

File: MovieApp/Screens/Search/SearchViewController.swift [hunk 2]
```diff
@@ -7,10 +7,146 @@
...
+		super.viewWillAppear(animated)
+		searchViewModel?.getMovies()
+		self.navigationController?.isNavigationBarHidden = false
+	}
+}
+
+// MARK: - Private function
+private extension SearchViewController {
+	func setupUI() {
+
```
Comment (namnh92) [inline MovieApp/Screens/Search/SearchViewController.swift]:
```suggestion

```

File: MovieApp/Screens/Search/SearchViewController.swift [hunk 3]
```diff
@@ -7,10 +7,146 @@
...
+		// MARK: - CollectionView
+		searchCollectionView.delegate = self
+		searchCollectionView.dataSource = self
+		searchCollectionView.register(UINib(nibName: ConstantSearch.searchCLVC, bundle: Bundle.main), forCellWithReuseIdentifier: ConstantSearch.searchCLVC)
+		searchCollectionView.register(UINib(nibName: ConstantSearch.searchDetailCell, bundle: Bundle.main), forCellWithReuseIdentifier: ConstantSearch.searchDetailCell)
+		searchCollectionView.register(UINib(nibName: ConstantSearch.searchCollectionReusableView, bundle: Bundle.main), forSupplementaryViewOfKind: UICollectionView.elementKindSectionHeader, withReuseIdentifier: ConstantSearch.searchCollectionReusableView)
```
Comment (namnh92) [inline MovieApp/Screens/Search/SearchViewController.swift:59]: Separate to sub function

File: MovieApp/Screens/Search/SearchViewController.swift [hunk 4]
```diff
@@ -7,10 +7,146 @@
...
+
+		// MARK: - SearchBar
+		searchController.searchBar.delegate = self
+		searchController.searchResultsUpdater = self
+		searchController.obscuresBackgroundDuringPresentation = false
+		searchController.searchBar.enablesReturnKeyAutomatically = false
+		navigationItem.searchController = searchController
+		navigationItem.hidesSearchBarWhenScrolling = false
```
Comment (namnh92) [inline MovieApp/Screens/Search/SearchViewController.swift:69]: Separate to sub function

File: MovieApp/Screens/Search/SearchViewController.swift [hunk 5]
```diff
@@ -7,10 +7,146 @@
...
+	}
+
+	func showError(_ errorMessage: String) {
+		// TODO: - Show popup error
+		print(errorMessage)
+	}
+}
+
+// MARK: - UICollectionViewDataSource
+extension SearchViewController: UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
```
Comment (namnh92) [inline MovieApp/Screens/Search/SearchViewController.swift]: Separate each extension only has one protocol

File: MovieApp/Screens/Search/SearchViewController.swift [hunk 6]
```diff
@@ -7,10 +7,146 @@
...
+
+	func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+			guard let cell = searchCollectionView.dequeueReusableCell(withReuseIdentifier: ConstantSearch.searchCLVC, for: indexPath) as? SearchCollectionViewCell else { return UICollectionViewCell() }
+			cell.configured(movieData: searchViewModel?.searchMovieViewModels[indexPath.row] )
+			return cell
+
+	}
+
+	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+		return CGSize(width: collectionView.frame.width, height: collectionView.frame.height / 6)
```
Comment (namnh92) [inline MovieApp/Screens/Search/SearchViewController.swift]: Make constants, don't use magic number *6*

File: MovieApp/Screens/Search/SearchViewController.swift [hunk 7]
```diff
@@ -7,10 +7,146 @@
...
+	}
+
+	func collectionView(_ collectionView: UICollectionView, viewForSupplementaryElementOfKind kind: String, at indexPath: IndexPath) -> UICollectionReusableView {
+		let headerView = collectionView.dequeueReusableSupplementaryView(ofKind: UICollectionView.elementKindSectionHeader, withReuseIdentifier: ConstantSearch.searchCollectionReusableView, for: indexPath) as! SearchCollectionReusableView
+		headerView.configure()
+		return headerView
+	}
+
+	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, referenceSizeForHeaderInSection section: Int) -> CGSize {
+		return CGSize(width: view.frame.size.width, height: view.frame.size.width / 10)
```
Comment (namnh92) [inline MovieApp/Screens/Search/SearchViewController.swift]: Make constants, don't use magic number 10

File: MovieApp/Screens/Search/Views/SearchMovieCell/SearchDetailCell.swift [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+class SearchDetailCell: UICollectionViewCell {
+
+	// MARK: IBOutlets
+	@IBOutlet weak var img: UIImageView!
+	@IBOutlet weak var searchName: UILabel!
+
+	// MARK: 
+	override func awakeFromNib() {
+		super.awakeFromNib()
+//		searchName.textColor = .white
```
Comment (namnh92) [inline MovieApp/Screens/Search/Views/SearchMovieCell/SearchDetailCell.swift]: Remove comment

**Review by namnh92 — CHANGES_REQUESTED** [review]

**Comment by namnh92** [comment]
Please create PR follow rule:
1. Name
2. Description

File: MovieApp/Screens/Search/SearchViewController.swift (hunk 4 above)
Comment (dong2810) [reply MovieApp/Screens/Search/SearchViewController.swift:69]: fixed

**Pull request merged**